Fix: Handle projects without floor plans in ROI report generation

roshan-1001 opened the pull request:
- Allow ROI reports to generate using project.min_price when no floor plans exist
- Remove unnecessary unit_size validation in V2 functions
- Fix null safety issues when accessing floorPlan properties
- Update both generateProjectROIReportServiceV2 and getProjectAIReportServiceV2

## Description

[Provide a brief description of the changes made in this pull request.]

## Related Issue

[If there is an associated issue, mention it here and provide a link.]

## Changes Made

[List the specific changes made in this pull request.]

## Screenshots

[If applicable, include screenshots or GIFs to visually demonstrate the changes.]

## Checklist

-   [ ] I have tested these changes locally.
-   [ ] I have added appropriate documentation or comments where necessary.
-   [ ] I have followed the coding style guidelines of this project.
-   [ ] I have updated the relevant documentation or README files.
-   [ ] I have reviewed my own code and ensured it is clean and readable.
-   [ ] I have assigned reviewers to this pull request.

## Additional Notes

[Add any additional notes or context that may be helpful for reviewers.]

roshan-1001 committed:
Fix: Handle projects without floor plans in ROI report generation

- Allow ROI reports to generate using project.min_price when no floor plans exist
- Remove unnecessary unit_size validation in V2 functions
- Fix null safety issues when accessing floorPlan properties
- Update both generateProjectROIReportServiceV2 and getProjectAIReportServiceV2

Kartik1110 commented on src/apps/user/project/project.service.ts:
```diff
@@ -1219,24 +1219,21 @@ export const generateProjectROIReportServiceV2 = async (
-
-    if (!floorPlan) {
-        throw new Error('Floor plan not found');
+        if (floorPlan) {
```
why do we need this if condition

roshan-1001 replied:
I was adding a warning message, also there is a typo !floorplan, removing this if statement

roshan-1001 committed:
Remove unnecessary warning logs from ROI report functions

- Remove warning log when floor plan is auto-selected in generateProjectROIReportServiceV2
- Remove warning log when floor plan is auto-selected in getProjectAIReportServiceV2
- Cleaner code without redundant logging
